int64 len
obj_registered[] objects_reg
